metloom/pointdata/geosphere_austria.py

```diff
@@ -0,0 +1,395 @@
+        search_geom = projected_geom.buffer(kwargs["buffer"])
+
+        # get metadata for all stations
+        all_meta = cls._retrieve_all_metadata()
+
+        # return empty collection if we didn't find any points
+        if all_meta is None:
+            return cls.ITERATOR_CLASS([])
+
+        # convert to a geodataframe
+        gdf = gpd.GeoDataFrame(
+            all_meta,
+            geometry=gpd.points_from_xy(
+                all_meta["lon"], all_meta["lat"],
+                all_meta["altitude"] * M_TO_FT
+            )
+        )
+        # TODO: is this correct?
+        gdf = gdf.set_crs("EPSG:4326")
+
+        # filter to points within shapefile
+        if kwargs['within_geometry']:
+            filtered_gdf = gdf[gdf.within(projected_geom.iloc[0]["geometry"])]
+        # filter to the overall bounding box
+        else:
+            box_df = shp_to_box(search_geom)
+            filtered_gdf = gdf[gdf.within(box_df.iloc[0]["geometry"])]
+
+        # filter to active stations
+        if kwargs["filter_to_active"]:
+            filtered_gdf = filtered_gdf.loc[filtered_gdf["is_active"] == "true"]
+
+        points = [
+            cls(row[0], row[1], metadata=row[2])
+            for row in zip(
+                filtered_gdf["id"],
+                filtered_gdf["name"],
+                filtered_gdf["geometry"],
+            )
+        ]
+        return cls.ITERATOR_CLASS(points)
+
+
+class GeoSphereCurrentPointData(GeoSpherePointDataBase):
+    """
+    Implement PointData methods for GeoSphere Austria data source
+    API documentation here
+    https://dataset.api.hub.geosphere.at/v1/docs/index.html
+    https://dataset.api.hub.geosphere.at/v1/docs/user-guide/resource.html
+
+    Datasets available here https://data.hub.geosphere.at/dataset/
+
+    We use tawes-v1 data which consists of data from the last 3 months
+    in 10 minute increment
+    """
+
+    ALLOWED_VARIABLES = GeoSphereCurrentVariables
+    URL = "https://dataset.api.hub.geosphere.at"
+    DATASOURCE = "GEOSPHERE"
+    META_EXTENSION = "/v1/station/current/tawes-v1-10min/metadata"
+
+    def _data_request(self, params):
+        """
+        Make get request and return JSON
+        Args:
+            params: dictionary of request parameters
+        Returns:
+            dictionary of response values
+        """
+        url = self.URL + "/v1/station/historical/tawes-v1-10min"
+        resp = requests.get(url, params=params)
+        resp.raise_for_status()
+        return resp.json()
+
+    def get_daily_data(
+        self,
+        start_date: datetime,
+        end_date: datetime,
+        variables: List[SensorDescription],
+    ):
+        """
+        See docstring for PointData.get_daily_data
+        Example query:
+        https://dataset.api.hub.geosphere.at/v1/station/current/
+        tawes-v1-10min?parameters=TL&station_ids=11035
+        """
+        return self._get_data(start_date, end_date, variables, "D")
```

Comment:
Do we want to stop queries > 3 months old here? E.g. what happens if my date ranges spans that time period? Do I get a partial request? Might be worth policing it before one waits to find out.

Reply:
Currently it will return null data for those time periods. I can throw a value error if end date is older than 3 months old instead

Reply:
I added a value error for end dates older than 3 months